src/kernels/kernelStartupCodeProvider.node.ts

```diff
@@ -25,6 +25,13 @@ export class KernelStartupCodeProvider implements IStartupCodeProvider {
     ) {}
 
     async getCode(kernel: IKernel): Promise<string[]> {
+        // If this is a live kernel, we shouldn't be changing anything by running startup code.
+        if (
+            !isPythonKernelConnection(kernel.kernelConnectionMetadata) &&
```

Comment:
Might be easy to forget to add this check when creating new providers. We could alternatively change the interface to `getCode` and `getCodeForLiveKernel` to be more explicit

Reply:
I don't think that's a good idea, the implementation should worry about the type of kernel we provide code for, after all the kernel is passed as an argument. 
Basically the implemetnation shouold worry about it.

i.e. if we have seprate interrfaces (seprate method) for each different implemtation then we end up with a single interface for every single enumeration which then makes it unnecessary to have an interface or even a base class.